src/main/java/com/demo/MybatisApplication/service/SubjectService.java

```diff
@@ -20,22 +22,35 @@
 public class SubjectService {
     @Autowired
     SubjectRepository subjectRepository;
-
     @Autowired
     SubjectMapper subjectMapper;
 
-    public SubjectEntityDisplayDto getSubjectById(@PathVariable Long id){
-        return subjectMapper.subjectEntityToSubjectDisplayDto(subjectRepository.findSubjectById(id));
+    public SubjectEntityDisplayDto getSubjectById(@NotNull Long id) {
+        SubjectEntity subjectEntity = subjectRepository.findSubjectById(id);
+        if (subjectEntity == null) {
+            throw new RuntimeException("Subject not found with id: " + id);
+        }
+        return subjectMapper.mapSubjectEntityToSubjectDisplayDto(subjectEntity);
     }
 
     public List<SubjectDisplayDto> getSubjects(){
-        List<SubjectDisplayDto> subjects = subjectMapper.EntitiesToDtos(subjectRepository.findAllSubjects());
+        List<SubjectDisplayDto> subjects = subjectMapper.mapSubjectEntitiesToDtos(subjectRepository.findAllSubjects());
         return subjects;
     }
 
     public List<SubjectAdditionDto> addSubjects(@RequestBody List<SubjectAdditionDto> subjects) {
-        List<SubjectEntity> subjectEntities = subjectMapper.SubjectAddtionDtosToSubjectEntities(subjects);
+        List<SubjectEntity> subjectEntities = subjectMapper.mapSubjectAddtionDtosToSubjectEntities(subjects);
         subjectRepository.saveSubjects(subjectEntities);
         return subjects;
     }
+
+    public ResponseEntity<String> assignSubjectsToStudentHandling(List<Long> subjectIds){
+        for (Long subjectId : subjectIds) {
+            SubjectEntity subjectEntity = subjectRepository.findSubjectById(subjectId);
+            if (subjectEntity == null) {
+                return ResponseEntity.badRequest().body("Subject with id " + subjectId + " not found");
+            }
+        }
+        return ResponseEntity.ok().body("Done");
```

Comment:
We should throw an exception and not return or construct the response body in service. 
The job of service is to just handle the business logic and not worry about response or status code.

Reply:
Okay